core-web/apps/dotcms-ui/src/app/portlets/dot-pages/dot-pages.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { Observable } from 'rxjs/internal/Observable';
+import { DotPagesState, DotPageStore } from './dot-pages-store/dot-pages.store';
+
+@Component({
+    selector: 'dot-pages',
+    templateUrl: './dot-pages.component.html',
+    styleUrls: ['./dot-pages.component.scss'],
+    providers: [DotPageStore]
+})
+export class DotPagesComponent {
+    vm$: Observable<DotPagesState> = this.store.vm$;
+
+    constructor(private store: DotPageStore) {
+        this.store.setInitialStateData();
+    }
+
+    /**
+     * Event to load more/less Favorite page data
+     *
+     * @param {boolean} areAllFavoritePagesLoaded
+     * @memberof DotPagesComponent
+     */
+    toggleFavoritePagesData(areAllFavoritePagesLoaded: boolean): void {
+        if (areAllFavoritePagesLoaded) {
+            this.store.limitFavoritePages();
+        } else {
+            this.store.loadAllFavoritePages();
+        }
```

Comment:
Why don't you make one method that receives the number of items? That way you can just use that in the request directly and is more scalable and less decoupled.

Reply:
done!